src/nvim/api/private/helpers.c

```diff
@@ -615,15 +615,34 @@ bool object_to_vim(Object obj, typval_T *tv, Error *err)
     case kObjectTypeBuffer:
     case kObjectTypeWindow:
     case kObjectTypeTabpage:
-    case kObjectTypeInteger:
-      if (obj.data.integer > INT_MAX || obj.data.integer < INT_MIN) {
+    case kObjectTypeInteger: {
```

Comment:
can't we just always store the handle as int64 in the `obj.data` union and then add casts as necessary? (most cases are already be handled by `gendispatch.lua`)